### DISCUSSION TITLE
Is there a way to save Wi-Fi passwords without binding the app?

### DISCUSSION THREAD

**Discussion by KokuraSona** [alienatedsec/yi-hack-v5#400] [Q&A] [answered]
I have a Yi Home camera (Chinese version) but cannot bind it to the app. I’ve tried firmware versions 0.4.1, 0.3.8, and even v4, but with no success. The camera connects to Wi-Fi after scanning the QR code but shows "invalid QR code." While the camera works and the website interface is accessible, it seems to reset and require re-binding whenever I restart it for any change. Is there a method to save my Wi-Fi password directly on the camera for automatic connection? I do not need the app's features. Thank you for your work

**Comment by alienatedsec** [comment]
If operated in China, you need a Chinese version of the YI app. If operated outside of China, you must use a proxy. Also, check #237 and #278 for some guidance

**Reply by KokuraSona** [reply]
In fact, I have already tried this. I attempted the Chinese app for Android, the overseas version from Google Play, the Chinese app for iOS, and even Xiaomi Home, but none worked. The camera keeps displaying "invalid QR code" and prompts me to reset the camera. Should I reset the camera? Would that help with the app pairing?

**Comment by KokuraSona** [comment, marked as answer]
I did it!
After encountering numerous issues, I finally found a solution. This method should only work for Yi Home cameras (Version: 17CN). I believe this issue only occurs with this specific version.

First, you need to download the official Chinese firmware from [here](https://www.yunyizhilian.com/products/camera.html), then format your SD card to FAT32 and place the firmware file in the root directory. Insert the SD card, power on the camera, and wait for it to boot up. If you hear a Chinese voice prompt, you’ve successfully restored the official firmware.

Next, I tried to bind the camera with Yi Home, but it failed. Some suggested using version 1.6.1 from 2015, but in my tests, this version no longer works. Eventually, I found that you can first pair the camera using Xiaomi Home, then open Yi Home (Chinese version) **and log in with the same Xiaomi account used for Xiaomi Home.** Yi Home will then be linked to the camera.

Finally, just follow the instructions to use the 0.3.8 firmware (I’ve only tested this version). After rebooting, the camera will automatically connect to Wi-Fi.
